build-support/githooks/pre-commit

```diff
@@ -65,6 +65,10 @@ if git rev-parse --verify "${MERGE_BASE}" &>/dev/null; then
   echo "* Checking types for targets marked \`type_checked\`"  
   ./build-support/bin/mypy.py || exit 1
 
+  if git diff "${MERGE_BASE}" --name-only | grep '\.py$' > /dev/null; then
+    echo "* Checking formatting of pyhon files"
+    ./build-support/bin/black.sh || die "To fix formatting, run \`build-support/bin/black.sh -f\`"
```

Comment:
I would move this closer to the top, right above the import check. It's likely to fail more than any other check and we want to fail quickly.